android/src/org/ppsspp/ppsspp/NativeActivity.java

```diff
@@ -911,6 +918,68 @@ public boolean onKeyUp(int keyCode, KeyEvent event) {
 			return NativeApp.keyUp(0, keyCode);
 		}
 	}
+	@Override
+	protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+		super.onActivityResult(requestCode, resultCode, data);
+
+		if (requestCode == RESULT_LOAD_IMAGE && resultCode == RESULT_OK && null != data) {
+			Uri selectedImage = data.getData();
+			String[] filePathColumn = { MediaStore.Images.Media.DATA };
+
+			Cursor cursor = getContentResolver().query(selectedImage,
+					filePathColumn, null, null, null);
+			cursor.moveToFirst();
+
+			int columnIndex = cursor.getColumnIndex(filePathColumn[0]);
+			String picturePath = cursor.getString(columnIndex);
+			cursor.close();
+			String imgName = "";
+			imgName =  picturePath.endsWith(".jpg")? "background.jpg":"background.png";
+			System.out.println(picturePath);
```

Comment:
Wouldn't it be better to remove the debugging lines now?

Also, it'd probably be cleaner if this...

```java
String imgName = "";
imgName =  picturePath.endsWith(".jpg")? "background.jpg":"background.png";
```

Was just:

```java
String imgName = picturePath.endsWith(".jpg") ? "background.jpg" : "background.png";
```

No need to set it to an empty string first when you're going to immediately set it to something else, right?

That said, it seems like copying files in Java is painful.  I wonder if it'd be better to send `bgImage_updated` the `picturePath` string, and then have the bgImage_updated code do...

```c++
		if (!value.empty()) {
			std::string dest = GetSysDirectory(DIRECTORY_SYSTEM) + (endsWithNoCase(value, ".jpg") ? "background.jpg" : "background.png");
			File::Copy(value, dest);
		}
		UIBackgroundShutdown();
		UIBackgroundInit(*uiContext);
```

That seems like it'd be simpler, and avoid the need for this slow looking copy function in Java.

-[Unknown]